be1-go/hub/base_channel.go

```diff
@@ -84,7 +84,15 @@ func (c *baseChannel) VerifyPublishMessage(publish message.Publish) error {
 
 	// Check if the structure of the message is correct
 	msg := publish.Params.Message
-	err := msg.VerifyAndUnmarshalData()
+
+	// Verify the data
+	err := c.hub.verifyJson(msg.RawData, DataSchema)
+	if err != nil {
+		return xerrors.Errorf("failed to validate the data: %v", err)
```

Comment:
Perhaps this should be `message.Error{}`?

Reply:
If I replace `%v` by `%w` the error can still be interpreted as an `message.Error{}`: [https://play.golang.org/p/S_QDU4Q0b4T](https://play.golang.org/p/S_QDU4Q0b4T). Should I do that?

Reply:
But the description of the error remains the same so it will not provide a long and complete description in the error message. So maybe it would be better to wrap it in `message.Error`.